extmod/modussl_mbedtls.c

```diff
@@ -237,6 +242,20 @@ STATIC mp_obj_ssl_socket_t *socket_new(mp_obj_t sock, struct ssl_args *args) {
         }
     }
 
+    if (args->cadata.u_obj != mp_const_none) {
+        size_t cacert_len;
+        const byte *cacert = (const byte *)mp_obj_str_get_data(args->cadata.u_obj, &cacert_len);
+        // len should include terminating null
+        ret = mbedtls_x509_crt_parse(&o->cacert, cacert, cacert_len + 1);
```

Comment:
Currently only DER format is supported.  And in such a case I think this function only parses the first DER in the buffer and will ignore anything after that.

To support a chain of DERs we would need to have a custom DER parser, or at least one that could tell the length of a DER so it could call `mbedtls_x509_crt_parse_der()` multiple times, once for each DER in the chain.

Reply:
True, I've just checked out `mbedtls_x509_crt_parse()` documentation:

> Parse one DER-encoded or one or more concatenated PEM-encoded certificates and add them to the chained list.
.... 
The buffer holding the certificate data in PEM or DER format. For certificates in PEM encoding, this may be a concatenation of multiple certificates; for DER encoding, the buffer must comprise exactly one certificate.

And `mbedtls_x509_crt_parse_der()`:
> Parse a single DER formatted certificate and add it to the end of the provided chained list.

Which seems like a concatenated DER-encoded certificates file does not exist. 

> custom DER parser, or at least one that could tell the length of a DER so it could call mbedtls_x509_crt_parse_der() multiple times, once for each DER in the chain

I guess you could parse multiple files/buffers calling `mbedtls_x509_crt_parse_der() ` multiple times, but from a single file I don't think is possible, you would need to add a header with the sizes or some delimiters like PEM. 

Probably is better to support PEM-encoding or make it multiple files/list of buffers than a single "chain DER file".
In any case I think this fits better in `SSLContext.load_verify_locations`, right?

Reply:
Let's solve this problem later on, as part of SSLContext.  For now what is done in this PR is a good start, and a subset of what we could eventually support (either PEM or list/tuple/concat'd DER).